### PR TITLE
Integrate rm with s3transfer

### PR DESCRIPTION
**Pull request by kyleknap**
This pull request integrates ``rm`` with ``s3transfer`` and the new architecture. All existing tests pass without any changes needed.

cc @jamesls @JordonPhillips

### CODE REVIEW

File: awscli/customizations/s3/results.py [hunk 1]
```diff
@@ -326,16 +341,19 @@ def _print_noop(self, **kwargs):
 
     def _print_success(self, result, **kwargs):
         success_statement = self.SUCCESS_FORMAT.format(
-            transfer_type=result.transfer_type, src=result.src,
-            dest=result.dest)
+            transfer_type=result.transfer_type,
+            direction=self._get_direction(result)
```
Comment (jamesls) [inline awscli/customizations/s3/results.py]: Can we rename `direction`?  "one way direction" sort of implies transfer in one direction, i.e local->s3.  Wasn't obvious to me what that meant.  What about something like `transfer_description`, `transfer_src_dest`, `transfer_locations`, etc.

File: awscli/customizations/s3/results.py [hunk 2]
```diff
@@ -186,7 +195,10 @@ def _get_ongoing_dict_key(self, result):
                 'Any result using _get_ongoing_dict_key must subclass from '
                 'BaseResult. Provided result is of type: %s' % type(result)
             )
-        return ':'.join([result.transfer_type, result.src, result.dest])
+        key_components = [result.transfer_type, result.src]
```
Comment (jamesls) [inline awscli/customizations/s3/results.py]: What about just `':'.join(str(el) for el in [...])` to avoid the `dest is None` special casing?

**Comment by jamesls** [comment]
Pretty straightforward, looks good to me.  Had a few questions/comments.

File: awscli/customizations/s3/results.py (hunk 2 above)
Comment (kyleknap) [inline awscli/customizations/s3/results.py]: Yeah that works. I will update that.

File: awscli/customizations/s3/results.py (hunk 1 above)
Comment (kyleknap) [inline awscli/customizations/s3/results.py]: I like ``transfer_location``. I will do that.

**Comment by kyleknap** [comment]
@jamesls it should be good to look at again. I think I addressed your feedback.

**Comment by jamesls** [comment]
:shipit:

**Pull request merged**